### DISCUSSION TITLE
ZX Interface 1 not working

### DISCUSSION THREAD

**Discussion by CaptHFlint** [UzixLS/zx-sizif-512#15] [Q&A]
I know Interface 1 isnt currently in the list of compatible addons but I would really like it to work.

I have tried multiple combinations of ROM settings whilst disabling DivMMC/SD card but still whenever I attach Interface 1 all I get is a screen of coloured boxes or sometimes coloured columns and I can't do anything. Reboot button causes the same problem. Magic button for some weired reason just has the word PAUSE flahing in the middle of the screen but no matter how long I wait nothing happens.

If anyone has an Interface 1 working with the Sizif 512 rev D then please let me know how you did it. Many thanks.

**Comment by UzixLS** [comment]
I just looked into Interface 1 schematic and - oh my god - it's using all possible voltages: +5V (of course), +12V (okay), +9V (hmm), -12VAC (omg). Sizif can only provide +5V and one of +12V or +9V (depending of your power supply and J3 jumper installed). And I don't know any clone providing all these voltages. Good news - looks like only +9V required for microdrive proper work.

Next issue - Interface 1 locking all odd ports! I don't think it would work with anything other than 48K speccy. Sizif using Pentagon mode by default - and - what is expected - can't switch to 48K mode at boot because all odd ports are locked - including those controlling Pentagon/128/+3/48 mode - whoops.

I prepared CPLD firmware witch boots into 48K mode by default - and it may work if we're lucky :)
[cpld.rev.D_20220824_48K_mode_by_default.pof.zip](https://github.com/UzixLS/zx-sizif-512/files/9418438/cpld.rev.D_20220824_48K_mode_by_default.pof.zip)

**Reply by CaptHFlint** [reply]
Thanks for such a quick response! 

FYI Interface 1 works on a 48k and a 128k Toastrack (not sure about others)

I am very new to this sort of thing so here comes a (probably dumb) question:
How would I use the zip to update the CPLD firmware?

**Reply by UzixLS** [reply]
Oh, I just forgot... 
There is two approach to lock ports. East (or Soviet, if you prefer) approach is to lock any possible port with IORQGE signal. That's more consistent and flexible way to do things. West (original) approach allows to lock only even ports, completely ignoring IORQGE for odd ports. That's makes all IORQGE concept is almost completely useless, but as benefit we get compatibility with some strange-engineered addons like Interface-1.
Sizif follows East way.

To update CPLD firmware you need USB Blaster (about ~5$ on ali) and Quartus Programmer software (full Quartus distribution isn't required).
Step-by-step manual:

0. Connect USB blaster to PC, install drivers if needed;
1. Connect USB blaster to JTAG header on Sizif board;
2. Power on Sizif;
3. Run Quartus Programmer;
4. Press "Auto Detect" button, EPM1270 should appears in window;
5. Double click on "< none >" entry in window, choose firmware file (cpld.rev.X.pof)
6. Check "Program" and "Verify" checkboxes;
7. Press "Start" button;
8. Wait for "100% Successfull";
9. Done :)
![изображение](https://user-images.githubusercontent.com/1556813/186500267-4b944c99-b8be-4664-997a-4db533a2fb78.png)

**Comment by CaptHFlint** [comment]
OK thanks I'll give it go!
